File: ai-job-portal-backend/recruiter-service/src/main/java/com/prahlad/aijobportal/recruiterservice/recruiter/service/RecruiterService.java
package com.prahlad.aijobportal.recruiterservice.recruiter.service;

import com.prahlad.aijobportal.recruiterservice.recruiter.dto.request.UpdateRecruiterProfileRequest;
import com.prahlad.aijobportal.recruiterservice.recruiter.dto.response.RecruiterResponse;

import java.util.UUID;

public interface RecruiterService {

    RecruiterResponse getMyProfile(UUID userId);

    RecruiterResponse updateMyProfile(UUID userId, UpdateRecruiterProfileRequest request);
}

File: ai-job-portal-frontend/src/features/recruiter-profile/services/recruiter.service.ts
import { apiClient } from "@/services/api-client";
import type { ApiResponse } from "@/types/api";
import type { RecruiterResponse, UpdateRecruiterProfileRequest } from "@/features/recruiter-profile/types";

/** Maps 1:1 to recruiter-service RecruiterController (/recruiter/profile). */
export const recruiterService = {
  getMyProfile: () =>
    apiClient.get<ApiResponse<RecruiterResponse>>("/recruiter/profile").then((res) => res.data),

  updateMyProfile: (payload: UpdateRecruiterProfileRequest) =>
    apiClient.put<ApiResponse<RecruiterResponse>>("/recruiter/profile", payload).then((res) => res.data),
};
